add docs for alternative markdown file extensions (#1893)
withastro/docs: src/pages/en/guides/markdown-content.md

i18nReady: true
---

> [Markdown](https://daringfireball.net/projects/markdown/) is commonly used to author text-heavy content like blog posts and documentation. Astro includes built-in support for standard Markdown (`.md` or `.markdown`) files. 

Reviewer: ```suggestion
[Markdown](https://daringfireball.net/projects/markdown/) is commonly used to author text-heavy content like blog posts and documentation. Astro includes built-in support for standard Markdown (`.md`, `.markdown`, '.mdown', '.mkdn', '.mkd', '.mdwn' and '.md'.) files. 
```

Changed my mind. Let's put it here like this!

With the [@astrojs/mdx integration](/en/guides/integrations-guide/mdx/) installed, Astro also supports [MDX](https://mdxjs.com/) (`.mdx`) files which bring added features like support for JavaScript expressions and components in your Markdown content. 

Use either or both types of files to write your Markdown content!

## Markdown and MDX Pages

> Astro treats any `.md`, `.markdown` or `.mdx` file inside of the `/src/pages/` directory as a page. Placing a file in this directory, or any sub-directory, will automatically build a page route using the pathname of the file.

Reviewer: ```suggestion
Astro treats any `.md` (or alternative supported extension) or `.mdx` file inside of the `/src/pages/` directory as a page. Placing a file in this directory, or any sub-directory, will automatically build a page route using the pathname of the file.
```

#### Example: Using one Layout for `.md`, `.mdx`, and `.astro` files

> A single Astro layout can be written to receive the `frontmatter` object from `.md`, `.markdown` and `.mdx` files, as well as any named props passed from `.astro` files.

Reviewer: ```suggestion
A single Astro layout can be written to receive the `frontmatter` object from `.md` (or `.markdown` etc.) and `.mdx` files, as well as any named props passed from `.astro` files.
```
What do you think about changing the rest of the occurances to this, now that we've listed at the top of the `markdown-content` page??  @MoustaphaDev

Author: Perfect 👍️

### Markdown Drafts

> `draft: true` is an optional frontmatter value that will mark an individual `.md` or `.markdown` page or post as "unpublished." By default, this page will be excluded from the site build.

Reviewer: ```suggestion
`draft: true` is an optional frontmatter value that will mark an individual `.md` (or `.markdown` etc.) page or post as "unpublished." By default, this page will be excluded from the site build.
```

### Variables and Components

:::caution[Deprecated]

> Astro v1.0 **only supports standard Markdown in `.md` and `.markdown` files**. The ability to use [components or JSX in Markdown pages is no longer enabled by default](/en/migrate/#deprecated-components-and-jsx-in-markdown) and support will eventually be removed entirely. 

Reviewer: ```suggestion
Astro v1.0 **only supports standard Markdown in `.md` (or `.markdown` etc.) files**. The ability to use [components or JSX in Markdown pages is no longer enabled by default](/en/migrate/#deprecated-components-and-jsx-in-markdown) and support will eventually be removed entirely. 
```

- [Astro components](/en/core-concepts/astro-components/) or [UI framework components](/en/core-concepts/framework-components/) in MDX (`.mdx`) files.

> See the migration guide for help [converting your existing Astro `.md` or `.markdown` files to `.mdx`](/en/migrate/#converting-existing-md-files-to-mdx).

Reviewer: ```suggestion
See the migration guide for help [converting your existing Astro `.md` ( or `.markdown` etc.) files to `.mdx`](/en/migrate/#converting-existing-md-files-to-mdx).
```

Astro comes with built-in support for [Shiki](https://shiki.matsu.io/) and [Prism](https://prismjs.com/). This provides instant syntax highlighting for:

> - all code fences (\`\`\`) used in a Markdown (`.md` or `.markdown`) or MDX (`.mdx`) file.

Reviewer: ```suggestion
- all code fences (\`\`\`) used in a Markdown (`.md` or `.markdown` etc.) or MDX (`.mdx`) file.
```
